naughty_strings_test.go

```diff
@@ -90,66 +92,43 @@ func TestNaughtyStrings(t *testing.T) {
 		NaughtyStrings(t, tv, tv.asSymbol()+"::"+tv.asSymbol()+"::"+tv.asSymbol()+"::"+tv.asString())
 	}
 
-	if err := scanner.Err(); err != nil {
-		t.Fatalf("expected scanner to scan without errors; %s", err.Error())
-	}
+	assert.NoError(t, scanner.Err(), "Expected scanner to scan without errors")
 }
 
 func NaughtyStrings(t *testing.T, tv testValue, s string) {
 	hasherProvider := newCryptoHasherProvider(SHA256)
 
 	str := strings.Builder{}
 	hw, err := NewHashWriter(ion.NewTextWriter(&str), hasherProvider)
-	if err != nil {
-		t.Fatalf("expected NewHashWriter() to successfully create a HashWriter; %s", err.Error())
-	}
+	require.NoError(t, err, "Expected NewHashWriter() to successfully create a HashWriter")
 
 	ionHashWriter, ok := hw.(*hashWriter)
-	if !ok {
-		t.Fatal("expected hw to be of type hashWriter")
-	}
+	require.True(t, ok, "Expected hw to be of type hashWriter")
 
 	writeToWriterFromReader(t, ion.NewReaderStr(s), ionHashWriter)
 
 	hr, err := NewHashReader(ion.NewReaderStr(s), hasherProvider)
-	if err != nil {
-		t.Fatalf("expected NewHashReader() to successfully create a HashReader; %s", err.Error())
-	}
+	require.NoError(t, err, "Expected NewHashReader() to successfully create a HashReader")
 
 	ionHashReader, ok := hr.(*hashReader)
-	if !ok {
-		t.Fatalf("expected hr to be of type hashReader")
-	}
+	require.True(t, ok, "Expected hr to be of type hashReader")
 
 	if !ionHashReader.Next() {
-		err = ionHashReader.Err()
-		if err != nil {
-			t.Errorf("expected ionHashReader.Next() to execute without errors; %s", err.Error())
-		}
+		assert.NoError(t, ionHashReader.Err(), "Something went wrong executing ionHashReader.Next()")
 	}
 
 	if !ionHashReader.Next() {
-		err = ionHashReader.Err()
-		if err != nil {
-			t.Errorf("expected ionHashReader.Next() to execute without errors; %s", err.Error())
-		}
+		assert.NoError(t, ionHashReader.Err(), "Something went wrong executing ionHashReader.Next()")
 	}
 
 	if tv.isValidIon() {
 		writerSum, err := ionHashWriter.Sum(nil)
-		if err != nil {
-			t.Fatalf("expected ionHashWriter.Sum(nil) to execute without errors; %s", err.Error())
-		}
+		require.NoError(t, err, "Something went wrong executing ionHashWriter.Sum(nil)")
 
 		readerSum, err := ionHashReader.Sum(nil)
-		if err != nil {
-			t.Fatalf("expected ionHashReader.Sum(nil) to execute without errors; %s", err.Error())
-		}
+		require.NoError(t, err, "Something went wrong executing ionHashReader.Sum(nil)")
 
-		if !reflect.DeepEqual(writerSum, readerSum) {
-			t.Errorf("expected reader/writer sums for \"%s\" to match;\nWriter sum: %v\nReader sum: %v",
-				tv.asIon(), writerSum, readerSum)
-		}
+		assert.Equalf(t, writerSum, readerSum, "Expected reader/writer sums for \"%s\" to match", s)
```

Comment:
What equality is the `assert` library using here? Is it the same as `reflect.DeepEqual`? 

I went to the repo for the `assert` lib but a quick look did not give me an answer.